### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by stdex** [paskalov/gender-recognition#1]
How about percentage of good recognition?
I'm try it and there are many wrong recognitions in it.
For example:
![angelina-jolie](https://cloud.githubusercontent.com/assets/4608879/8493581/eacf2f54-2164-11e5-88e0-cf772f00c7c0.jpg)
![facedetection](https://cloud.githubusercontent.com/assets/4608879/8493583/01accf60-2165-11e5-9294-f6f274f555c6.png)
![mika-nakashima04](https://cloud.githubusercontent.com/assets/4608879/8493592/1d89a424-2165-11e5-8f36-95ae27ffa886.jpg)
![ekavaz 6](https://cloud.githubusercontent.com/assets/4608879/8493598/427c6596-2165-11e5-9ef2-39ef4a77f29f.jpg)
![asamma 1](https://cloud.githubusercontent.com/assets/4608879/8493606/548a9ec4-2165-11e5-9c61-dbba783560a4.jpg)

**Comment by paskalov** [comment]
Hello. How do you present these pictures to the neural network?

**Comment by stdex** [comment]
> private static File file;
> ...
> Mat image = Imgcodecs.imread(file.getPath());
> ...
> MatOfByte byteMat = new MatOfByte();
> Imgcodecs.imencode(".jpg", image, byteMat);
> byte[] bytes = byteMat.toArray();
> BufferedImage img = ImageIO.read(new ByteArrayInputStream(bytes));
> ...
> GenderRecognition genderOutput = new GenderRecognition();
> String gender = genderOutput.recognize(img);

**Comment by paskalov** [comment]
In order to present your image to the neural network, you must detect face on image, resize detected face to 125x127 pixels and then grayscale it. Valid sample image for the neural network attached below
![woman_32](https://cloud.githubusercontent.com/assets/9070571/8497441/9ba465de-217d-11e5-9bb5-d56d9b9fa778.jpg)
P.S: reference for all the steps mentioned above you can find in source file VideoCam.java.

**Comment by stdex** [comment]
Oh, sorry. However, there were problems with recognition.
![image](https://cloud.githubusercontent.com/assets/4608879/8498401/0dfb508c-218f-11e5-826a-cec5558a7b1f.png)
![image](https://cloud.githubusercontent.com/assets/4608879/8498409/20c510cc-218f-11e5-86e8-b80629460422.png)
![image](https://cloud.githubusercontent.com/assets/4608879/8498433/33a18e28-218f-11e5-98bf-d60d9cef179d.png)
![image](https://cloud.githubusercontent.com/assets/4608879/8498439/48062482-218f-11e5-9073-0d5f7d742b8c.png)
![image](https://cloud.githubusercontent.com/assets/4608879/8498445/5fd8af08-218f-11e5-9650-995c69749647.png)

I'm use swing wrapper, for load images, code and image datasets here:
https://github.com/stdex/FaceRecognition

**Comment by paskalov** [comment]
Thank you for pointing this out. I had some similar outputs when I tested this neural network. Problem is in training set. I had only 294 faces of woman and man respectively to train the neural network, learning rate was 0.2 and momentum was 0.7; total neural network error is 0.01. What I've discovered, that output of this neural network depends mostly on picture's brightness, facial hair (not only beards and moustache, but bangs). If you want more precise output, you need to train a neural network with more diverse data set. 
P.S: It's a good thing using histogram equalization on image before training, but with mine data set it didn't work out very well.

**Comment by stdex** [comment]
In due time, I was looking for the face datasets, maybe it will help to improve the neural network.
1. The Japanese Female Facial Expression (JAFFE) Database
   213 images, 10 individuals, 256x256, grayscale
   http://www.kasrl.org/jaffe.html
2. FEI Face Database
   200 images, 100 individuals, 260x360, RGB
   200 images, 100 individuals, 250x300, grayscale
   200 images, 100 individuals, 162x193, grayscale
   http://fei.edu.br/~cet/facedatabase.html
3. The Extended Yale Face Database B [now not avaliable]
   16128 images, 28 individuals, 640x480, grayscale, different illumination
   16128 images, 28 individuals, 168x192, grayscale, different illumination
   http://vision.ucsd.edu/~iskwak/ExtYaleDatabase/ExtYaleB.html
4. The Sheffield (previously UMIST) Face Database
   564 images, 20 individuals, 92x112, grayscale, sets at different rotations of the head
   https://www.sheffield.ac.uk/eee/research/iel/research/face
5. Utdallas Face Database
   Many sets by ~150 images.
   http://agingmind.utdallas.edu/facedb
6. The ORL Database of Faces
   20 individuals, 92x112, grayscale, sets at different rotations of the head.
   http://www.cl.cam.ac.uk/research/dtg/attarchive/facedatabase.html
7. Labeled Faces in the Wild Home
   13233 images, 5749 individuals
   http://vis-www.cs.umass.edu/lfw/
8. The MUCT Face Database
   3755 images
   http://www.milbo.org/muct/
9. BioID Face Database
   1521 images, 23 individuals, 384x286
   https://www.bioid.com/About/BioID-Face-Database
10. Faces in the Wild
    30281 faces collected from News Photographs
    http://tamaraberg.com/faceDataset/index.html
11. Adience collection of unfiltered faces for gender and age classification
    26580 images, **with gender labels**
    http://www.cslab.openu.ac.il/download/
    Login: adiencedb
    Password: adience
12. Dr Libor Spacek Collection of Facial Images
    7900 images, 395 individuals, 180x200, **with gender labels**
    http://cswww.essex.ac.uk/mv/allfaces/index.html
13. Frontal face dataset. Collected by Markus Weber at California Institute of Technology.
    450 images, 27 individuals, 896x592
    http://www.vision.caltech.edu/html-files/archive.html
14. Georgia Tech face database
    19786 images, 50 individuals, 150x150
    http://www.anefian.com/research/face_reco.htm
15. YouTube Faces DB
    3,425 videos of 1,595 different people
    ftp://agas.openu.ac.il/
    Login: wolftau
    Password: wtal997
16. FaceScrub dataset
    107818 images, 530 individuals, **with gender labels** 
    http://vintage.winklerbros.net/faceScrub.zip
    Password: ICIP'14
17. PubFig: Public Figures Face Database
    58,797 images of 200 people
    http://www.cs.columbia.edu/CAVE/databases/pubfig/
18. The Labeled Wikipedia Faces
    8.5k faces for about 1.5k identities
    http://www.info.polymtl.ca/~mdhas/LWF/
19. Indian Movie Face database
    34512 images of 100 Indian actors, **with gender labels** 
    http://cvit.iiit.ac.in/projects/IMFDB/#
20. 2D face sets
    Some small sets
    http://pics.psych.stir.ac.uk/2D_face_sets.htm

**Comment by paskalov** [comment]
Thank you a lot! I'll try to train new network and I'll commit it to this repository later.
